crates/turborepo-lib/src/run/summary/execution.rs

```diff
@@ -1,28 +1,164 @@
 use std::fmt;
 
-use chrono::{DateTime, Duration, Local};
-use serde::{ser::SerializeStruct, Serialize};
+use chrono::{DateTime, Duration, Local, SubsecRound};
+use serde::{ser::SerializeStruct, Serialize, Serializer};
 use tokio::sync::mpsc;
-use turbopath::AnchoredSystemPath;
+use tracing::log::warn;
+use turbopath::{AbsoluteSystemPathBuf, AnchoredSystemPath};
+use turborepo_ui::{color, BOLD, BOLD_GREEN, BOLD_RED, MAGENTA, UI};
 
-use crate::run::task_id::TaskId;
+use crate::run::{summary::task::TaskSummary, task_id::TaskId};
 
 // Just used to make changing the type that gets passed to the state management
 // thread easy
 type Message = Event;
 
+fn serialize_datetime<S: Serializer>(
+    date_time: &DateTime<Local>,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    serializer.serialize_i64(date_time.timestamp_millis())
+}
+
+fn serialize_optional_datetime<S: Serializer>(
+    date_time: &Option<DateTime<Local>>,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    let millis = date_time
+        .map(|dt| dt.timestamp_millis())
+        .unwrap_or_default();
+    serializer.serialize_i64(millis)
+}
+
 // Should *not* be exposed outside of run summary module
 /// The execution summary
 #[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
 pub struct ExecutionSummary<'a> {
     // this thread handles the state management
     #[serde(skip)]
     state_thread: tokio::task::JoinHandle<SummaryState>,
     #[serde(skip)]
     sender: mpsc::Sender<Message>,
     command: String,
-    package_inference_path: Option<&'a AnchoredSystemPath>,
-    started_at: DateTime<Local>,
+    #[serde(rename = "repoPath", skip_serializing_if = "Option::is_none")]
+    package_inference_root: Option<&'a AnchoredSystemPath>,
+    #[serde(serialize_with = "serialize_datetime")]
+    pub(crate) start_time: DateTime<Local>,
+    #[serde(serialize_with = "serialize_optional_datetime")]
+    pub(crate) end_time: Option<DateTime<Local>>,
+    pub(crate) exit_code: Option<u32>,
+}
+
+impl ExecutionSummary<'_> {
+    fn duration(&self) -> String {
+        let duration = self
+            .end_time
+            .unwrap_or_else(Local::now)
+            .trunc_subsecs(3)
+            .signed_duration_since(self.start_time.trunc_subsecs(3));
+
+        if duration.num_hours() > 0 {
+            format!(
+                "{}h{}m{}s",
+                duration.num_hours(),
+                duration.num_minutes(),
+                duration.num_seconds()
+            )
+        } else if duration.num_minutes() > 0 {
+            format!("{}m{}s", duration.num_minutes(), duration.num_seconds())
+        } else if duration.num_seconds() > 0 {
+            format!("{}s", duration.num_seconds())
+        } else {
+            format!("{}ms", duration.num_milliseconds())
+        }
+    }
```

Comment:
I think this can get extracted to something like and it would decouple our display logic for time from the rest of the summary
```
struct TurboTime(OldDuration)

impl TurboTime {
    pub fn new(start: DateTime, end: DateTime) -> Self {
        Self(end_time.trunc_subsecs(3).singed_duration_since(start.truc_subset(3)))
    }
}

impl fmt::Display for TurboTime {
...
}
```